subprojects/docs/src/docs/userguide/taskConfiguration.adoc

```diff
@@ -0,0 +1,269 @@
+class org.gradle.api.plugins.quality.CodeNarc 2
+----
+1. 188 tasks are created with the old API and 587 tasks are registered with the new API for a total of 775 tasks.
+2. 613 tasks were created and configured and only 162 tasks were avoided (never created or configured). 21% of all tasks were avoided (higher is better).
+3. Lists of the type of tasks that were created with the old API. This is a good list to work down to increase the amount of possible avoidable task configuration. 
+4. Lists of the type of tasks that were created with the new API but were created/configured anyways. This is a good list to work down to increase the amount of task configuration that is avoided.
+
+These statistics are printed out once per build. So projects with buildSrc will display this information twice (once for buildSrc, once for the root build).  In a build that uses the new APIs perfectly, we should see 0 tasks created with the old API and only 1 created/configured task because we are only executing the `help` task. If you run other tasks (like `build`), you should expect many more tasks to be created and configured.
+
+You can use the list of task types to guide which tasks would provide the biggest bang for your buck when you migrate to the new API.
+
+To approximate the time it takes to configure a build without executing tasks, you can run `gradle help`. Please use the Gradle Profiler to measure your build as described in <<sec:new_task_gradle_profiler_scenario>>.
+
+[[sec:lazy_task_pitfalls]]
+==== Pitfalls and examples
+
+The following guidelines should be followed when using the new APIs:
+
+- Migrate task configuration that affects all tasks TODO (`tasks.all {}`) or subsets by type (`tasks.withType(...) {}`). This will cause your build to create fewer built-in Gradle task types.
+- Migrate tasks that are configured by name. 
+- Migrate tasks created in custom plugins to use TODO `register(...)` and `configureEach {}`.
+- Use the information about which task types are created or use the old API to guide which build scripts or plugins to modify. 
+- When inside a task configuration action, avoid configuring anything other than that task.  This includes other tasks and extensions. 
+
+[[sec:old_vs_new_configuration_api_overview]]
+=== Old vs New API overview
+
+[NOTE]
+====
+Methods that take a `groovy.lang.Closure` are covered in the new API with methods taking `org.gradle.api.Action`.
+====
+
+[cols="a,a", options="header"]
+|===
+| Old API | New API
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.util.Map)[]
+| No direct equivalent.
+2+| Use one of the alternatives below.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.util.Map,groovy.lang.Closure)[]
+| No direct equivalent.
+2+| Use one of the alternatives below.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.lang.String)[]
+| api:org.gradle.api.tasks.TaskContainer#register(java.lang.String)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
```

Comment:
on the generated html page, the formatting seems to be off here - html table extends off the background div, "this returns a taskprovider" appears under the `tasks.create()` cell.